src/net/sourceforge/kolmafia/combat/Macrofier.java

```diff
@@ -610,4 +626,25 @@ public static void macroManaRestore(StringBuffer macro) {
     macro.append("abort \"No MP restoratives!\"\n");
     macro.append("mark mafiampexit\n");
   }
+
+  private static class MacroOverride {
+    public MacroOverride(String macroOverride, ScriptRuntime macroInterpreter) {
+      this.macroOverride = macroOverride;
+      this.macroInterpreter = macroInterpreter;
+      setJavaScriptVariables(Macrofier.macroFunction, Macrofier.macroScope, Macrofier.macroThisArg);
+    }
+
+    public void setJavaScriptVariables(
```

Comment:
Since this is only used in the constructor, I don't see why it couldn't be inline there.
Perhaps preceded with ```// JavaScript variables```
But I'm not going to require it.

Reply:
good idea, thanks!